video.go

```diff
@@ -11,11 +11,13 @@ import (
 )
 
 type Video struct {
-	ID       string
-	Streams  []Stream
-	Title    string
-	Author   string
-	Duration time.Duration
+	ID              string
+	Streams         []Stream
+	Formats         []Format
+	AdaptiveFormats []AdaptiveFormat
```

Comment:
Streams, Formats and AdaptiveFormats seems redundant for me. I'll try to refactor this.